### PR TITLE
[ENH] set seed in test_score

### PR DESCRIPTION
**Pull request by Xinyu-Wu-0000**
This PR add seed setting in `test_score` in `TestAllForecasters`. 

The `DeepAR` model in PR #6551 is giving different output every call to `predict` which causes problem with `test_score`.

As suggested by @benHeid, maybe we can pass some parameters to the `predict` function in pytorch-forecasting model to control the sampling behavior, howerver I can't find such parameters. Luckily after setting seed for some fundamental packages before every call to `predict`, DeepAR gives the same output every time. 

As mentioned by @fkiraly, more than one deep learning models are facing the same problem. Setting seed in `test_score` for some fundamental packages could be helpful for models using sampling or other non-deterministic methods to generate prediction.

### CODE REVIEW

**Comment by Xinyu-Wu-0000** [comment]
As discussed in the tech session, we might need to reset the seed back to original seed. We can do it for `numpy`, `random` and `torch` but not `tensorflow`.

**Review by fkiraly — CHANGES_REQUESTED** [review]
To quickly summarize my feedback from the tech session:

* we need to make sure that setting of seeds is not global and affects other tests or estimators
* `numpy.seed` is afaik deprecated, generators should be used instead. This works via `random_state` for model that support it. Most deep learning models do not.
* related to the above but not this PR directly, most tensorflow based models fail tests that also assume we can derandomize, e.g., `test_multiprocessing_idempotent`, hence this test is skipped for them in `tests._config`. If we can find a way to set the seed on a per-estimator-instance basis for tf or torch, that would be great.
